tests/test_siglent_ssa3000x_spectrum_analyzer_data_client.py

```diff
@@ -0,0 +1,88 @@
+# it under the terms of the GNU General Public License as published by
+# the Free Software Foundation, either version 3 of the License, or
+# (at your option) any later version.
+#
+# This program is distributed in the hope that it will be useful,
+# but WITHOUT ANY WARRANTY; without even the implied warranty of
+# MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+# GNU General Public License for more details.
+#
+# You should have received a copy of the GNU General Public License
+# along with this program.  If not, see <https://www.gnu.org/licenses/>.
+
+import collections
+import contextlib
+import logging
+import types
+import unittest
+
+import numpy as np
+from lsst.ts import salobj, utils
+from lsst.ts.ess import csc
+
+logging.basicConfig(
+    format="%(asctime)s:%(levelname)s:%(name)s:%(message)s", level=logging.DEBUG
+)
+
+
+class DataClientTestCase(unittest.IsolatedAsyncioTestCase):
+    def setUp(self) -> None:
+        salobj.set_random_lsst_dds_partition_prefix()
+        config_schema = (
+            csc.SiglentSsa3000xSpectrumAnalyzerDataClient.get_config_schema()
+        )
+        validator = salobj.DefaultingValidator(config_schema)
+        config_dict = dict(
+            host="localhost",
+            port=5000,
+            connect_timeout=5,
+            read_timeout=1,
+            location="Test Location",
+            sensor_name="MockSSA3000X",
+            sleep_interval=0.1,
+            start_frequency="0.0 GHz",
+            stop_frequency="3.0 GHz",
+        )
+        config_dict = validator.validate(config_dict)
+        self.config = types.SimpleNamespace(**config_dict)
+
+    @contextlib.asynccontextmanager
+    async def create_controller(self) -> collections.abc.AsyncGenerator:
+        """Create an ESS Controller and Remote and wait for them to start."""
+        index = next(utils.index_generator())
+        async with salobj.Controller(
+            name="ESS", index=index
+        ) as self.controller, salobj.Remote(
+            domain=self.controller.domain, name="ESS", index=index, readonly=True  # type: ignore
+        ) as self.remote:
+            yield
+
+    async def test_siglent_ssa3000x_spectrum_analyzer_data_client(self) -> None:
+        async with self.create_controller():
+            data_client = csc.SiglentSsa3000xSpectrumAnalyzerDataClient(
+                config=self.config,
+                topics=self.controller,
+                log=self.controller.log,
+                simulation_mode=1,
+            )
+
+            await data_client.start()
+            try:
+                telemetry = await self.remote.tel_spectrumAnalyzer.next(flush=False)
+                np.testing.assert_almost_equal(
+                    telemetry.frequency,
+                    csc.SiglentSsa3000xSpectrumAnalyzerDataClient.frequencies,
+                )
+                assert np.amax(telemetry.trace) <= 0.0
+                assert np.amin(telemetry.trace) >= -100.0
```

Comment:
I would prefer to see the test be made much more specific:
* I think you can predict what the frequencies should be, and if so, please do so.
* Can you predict what the "trace" (spectrum) itself will look like? If so, it would be better to check that. Why are the value all negative?

Reply:
It already checked the frequencies and the test makes sure they are the same as the input. I modified it now because of the changes to ts_xml to check the start and stop frequencies.

As for the spectrum (previously trace) values, those are randomly generated. They need to be negative since it is measuring values in dB, which always are negative values.